tests/datasource/fluent/integration/test_sql_datasources.py

```diff
@@ -425,5 +526,267 @@ def test_checkpoint_run(
         assert result.success is True
 
 
+# TODO: remove items from this lookup when working on fixes
+REQUIRE_FIXES: Final[dict[str, list[DatabaseType]]] = {
+    'expect_column_values_to_not_be_null-str "lower"': ["postgres", "snowflake"],
+    "expect_column_values_to_not_be_null-str LOWER": [
+        "databricks_sql",
+        "postgres",
+        "snowflake",
+        "sqlite",
+    ],
+    "expect_column_values_to_not_be_null-str upper": [
+        "databricks_sql",
+        "postgres",
+        "sqlite",
+    ],
+    'expect_column_values_to_not_be_null-str "UPPER"': ["postgres", "snowflake"],
+    "expect_column_values_to_not_be_null-str UPPER": ["snowflake"],
+    "expect_column_values_to_not_be_null-quoted_name UPPER quote=None": ["snowflake"],
+    "expect_column_values_to_not_be_null-quoted_name UPPER quote=True": ["snowflake"],
+    "expect_column_values_to_not_be_null-quoted_name UPPER quote=False": ["snowflake"],
+    "expect_column_values_to_not_be_null-quoted_name upper quote=None": [
+        "databricks_sql",
+        "postgres",
+        "sqlite",
+    ],
+    "expect_column_to_exist-str LOWER": [
+        "databricks_sql",
+        "postgres",
+        "snowflake",
+        "sqlite",
+    ],
+    "expect_column_to_exist-str upper": ["databricks_sql", "postgres", "sqlite"],
+    "expect_column_to_exist-str UPPER": ["snowflake"],
+    'expect_column_to_exist-str "lower"': ["postgres", "snowflake", "sqlite"],
+    'expect_column_to_exist-str "UPPER"': ["postgres", "snowflake", "sqlite"],
+    "expect_column_to_exist-quoted_name UPPER quote=False": ["snowflake"],
+    "expect_column_to_exist-quoted_name UPPER quote=True": ["snowflake"],
+    "expect_column_to_exist-quoted_name UPPER quote=None": ["postgres", "snowflake"],
+    "expect_column_to_exist-quoted_name upper quote=None": [
+        "databricks_sql",
+        "postgres",
+        "sqlite",
+    ],
+}
```

Comment:
This is a mapping of all the column identifier tests that currently fail, the values are the dialects that it fails for.
It's being used to xfail the tests so that this PR can merge.
We can use this as a punch list for what needs to be fixed.